app/src/org/commcare/android/adapters/EntityListAdapter.java

```diff
@@ -225,7 +225,13 @@ private void search() {
             long startTime = System.currentTimeMillis();
             //It's a bit sketchy here, because this DB lock will prevent
             //anything else from processing
-            SQLiteDatabase db = CommCareApplication._().getUserDbHandle();
+            SQLiteDatabase db;
+            try {
+                db = CommCareApplication._().getUserDbHandle();
+            } catch (SessionUnavailableException e) {
+                this.finish();
+                return;
+            }
```

Comment:
Not sure how to handle this exception, since it is running in a launched thread.
Does this look right?